### ISSUE TITLE
[Potential bug] Grouping changes x values in unexpected way

### ISSUE THREAD

**Issue by zimmer-m** [wandb/client#1566]
**Describe the bug**
**What I do is the following**: I train a model for 200 epochs and log several metrics (like train.accuracy, test.accuracy..) using wandb.log and everything works as expected. I do not pass a different step argument than the default. Then after training, I save the model and loop over a list of desired sparsities: self.desired_sparsities = [0, 0.3, 0.5, 0.7, 0.9, 0.95, 0.98]. For each of the sparsities, I jump back to the saved model checkpoint, remove as many parameters as desired (e.g. 0,3 = 30%), finetrain the pruned model (without logging anything to wandb). After finetuning it, I use wandb.log to log the current desired_sparsity and training metrics using a different name (e.g. finetune.train.accuracy, finetune.test.accuracy). Then I redo the same steps for the next sparsity in the list. Hence, the new metrics finetune.test.accuracy as well as desired_sparsity are associated with step =201, 202, 203, ... 

**What I want to achieve**: My goal is then to have a line plot in the webinterface which shows desired_sparsity on the x-axis and finetune.test.accuracy on the y axis. I simply add the plot via the Webinterface. This works just fine without grouping of runs, but as soon as I group them together to analyze different hyperparameters, I get weird behaviour: the x-values (i.e. desired_sparsity) seem to allow values greater than 1, even though I fixed the list of values above (see pictures below). 

**Expected behavior**
I expect to see a correct plot finetune.test.accuracy vs. desired_sparsity, even when I group the runs. When grouping, I'd expect to see some sort of average over the groups, even in this plot. With sparsities between 0 and 1 it is not possible to obtain values greater than 1. Grouping should yield a plot of average finetune.test.accuracy's for each step/desired_sparsity.

**Screenshots**
- The expected and good result: ![image](https://user-images.githubusercontent.com/66726489/100843241-5dd7a480-347a-11eb-947e-96b985b65c28.png)

- The unexpected and **bad** result after grouping: 
![image](https://user-images.githubusercontent.com/66726489/100843473-b5761000-347a-11eb-9c9a-9837041dbada.png)

**Operating System**
 - OS: Ubuntu 20.04
 - Browser: Firefox
 - Version: 83

**Additional context**
I hope the above was not diffuse and it was understandable what my problem is, if not feel free to ask. This issue is related to the workaround I proposed here https://github.com/wandb/client/issues/1557#issue-753345198.

Thanks!

**Comment by zimmer-m** [comment]
Not sure if this is of any help, but I was able to circumvent the problem by plotting finetune.test.accuracy vs step (instead of vs. desired_sparsity). Then I selected desired_sparsity as the X-Axis-Expression, and not the values are at least correct. It would be nice of someone could elaborate on what is happening here, as I seem to have some misunderstandings of the core functionality of lineplot.

How it looks now after grouping:
![image](https://user-images.githubusercontent.com/66726489/100863982-a781b880-3495-11eb-90c4-21df4ad48181.png)

Can't find a way to scale the x-axis to [0,1] without destroying the plot again.

**Comment by tyomhak** [comment]
Hey @zimmer-m , thanks for writing in. @cvphelps can you please help here?

**Comment by ariG23498** [comment]
Hey @zimmer-m 
I was able to reproduce your issue.
## The code
```python
import wandb
import math
import random
for runs in range(10):
  run = wandb.init(project="xxx",entity="xxx")
  for step in range(10):
    wandb.log({
        "train.acc": math.log(.1 + random.random() + step * 0.01),
        "test.val_acc": math.log(.1 + random.random() + step *0.01)
        })
  desired_sparsities = [0, 0.3, 0.5, 0.7, 0.9, 0.95, 0.98]
  for step in desired_sparsities:
    wandb.log({
        "finetune.train.acc": math.log(.1 + random.random() + step * 0.01),
        "finetune.train.val_acc": math.log(.1 + random.random() + step *0.01),
        "desired_sparsities": step
        })
  run.finish()
```

This is a minimal implementation of your issue. I am considering 10 epochs and then later iterating over the `desired_sparsities`.

## Web-interface
The link to the project page: https://wandb.ai/repro/grouping

**My observations**:
- Line Plots are perfectly fine without grouping. I have made two kinds of line plots, one where the `x-axis` comes from `desired_sparsities` and the other where it comes from `step`. As expected, both the charts have their dissimalirites quite evident. On the one hand `step` lies in the range of `[10-16]` and on the other hand `desired_sparsities` iterate over `= [0, 0.3, 0.5, 0.7, 0.9, 0.95, 0.98]`.
- Line Plots plays funny with grouping. The problem for me was to be confused in the two fields namely `X` and `X Axis Expression`.
![image](https://user-images.githubusercontent.com/36856589/101124284-ba72c500-361c-11eb-8f2d-05a90e605086.png)
![image](https://user-images.githubusercontent.com/36856589/101124329-ce1e2b80-361c-11eb-841e-c21c5e861356.png)

These play an interesting role in the grouping mechanism. As @zimmer-m has mentioned, with `X=Step` and `X Axis Expression=desired_sparsities` we get the plots as we would expect, where the x axis has values corresponding `desired_sparsity` list.
![image](https://user-images.githubusercontent.com/36856589/101124527-4389fc00-361d-11eb-89dc-75f2a802cab0.png)

But here we cannot specify the range of the x values.

**Comment by zimmer-m** [comment]
> Hey @zimmer-m
> I was able to reproduce your issue.
Thanks for the MWE, appreciate it!

**Comment by brett-daley** [comment]
I have the same issue. I'm plotting a certain metric every 5000 steps. When the runs aren't group, the x-values are correct:
![MicrosoftTeams-image (8)](https://user-images.githubusercontent.com/20010659/123689299-11422500-d821-11eb-82bb-4c8f35de7019.png)
But when I group them, the x-values change to fractional values that don't align with the 5000 tick marks:
![MicrosoftTeams-image (9)](https://user-images.githubusercontent.com/20010659/123689364-25862200-d821-11eb-937a-fb57be48a06e.png)
I can't tell if this is a bug, or if it's intentional behavior from wandb (e.g. an interpolation strategy to efficiently implement the average when there are many runs grouped together). It would be nice to see some documentation about this.

**Comment by AntonioLopardo** [comment]
Same problem over here, 0-1 metric used as X-axis hallucinates a value for 1.005, are there any updates on this?

**Comment by dishank-b** [comment]
@cvphelps is there any update on this?

**Comment by htahboub** [comment]
+1, facing the same bug with some grouped runs.
